DoorGate/DoorPresenter.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  DoorPresenter.swift
+//  
+//
+//  Created by Yuliani Noriega on 1/26/18.
+//
+
+struct DoorPresenter {
+    let doorEntity = DoorEntity()
+    
+    mutating func buttonTap(command:DoorCommand) {
```

Comment:
what is the mutating modifier?

Reply:
you need to declare mutating when modifying variables in a struct, you see this in line 15 doorEntity.state.value += 1. 

It's something the compiler helps you with when you start modifying values in a struct